File: Exavision.Seasense.Shared/Capabilities/Camera/CameraZoomContinuousCapability.cs
using Exavision.Seasense.Shared.Settings;

namespace Exavision.Seasense.Shared.Capabilities.Camera {
    public abstract class CameraZoomContinuousCapability : Capability<SettingCapabilityEmpty>, ICameraZoomContinuousCapability {
        public abstract void ZoomInStart();

        public abstract void ZoomOutStart();

        public abstract void ZoomStop();

        public override SettingCapabilityEmpty GetSetting(SettingCapabilityEmpty setting) {
            setting.CapabilityType = CapabilityType.CameraZoomContinuous;
            return setting;
        }
    }
}

File: Exavision.Seasense/Client/src/app/materials/capabilities/camera/camera-zoom-continuous-capability.ts
import { Material } from "../../material";
import { SettingCapability } from "../../settings/setting-capability";
import { Site } from "../../site";
import { CapabilityState } from "../../states/capability-state";
import { Unit } from "../../unit";
import { Capability } from "../capability";
import { CapabilityType } from "../capability-type";

export class CameraZoomContinuousCapability implements Capability {
 
  
  id: string;
  capabilityType: CapabilityType;
  material: Material | undefined;


  constructor(settingCapability: SettingCapability) {
    this.id = settingCapability.id;
    this.capabilityType = CapabilityType.CameraZoomContinuous;
  }

  zoomOutStart() {
    if (this.material == null) return;
    if (this.material.unit == null) return;
    this.material?.wsService.cameraZoomOutStart(this.material.unit.id, this.material.id);


  }
  zoomStop() {
    if (this.material == null) return;
    if (this.material.unit == null) return;
    this.material?.wsService.cameraZoomStop(this.material.unit.id, this.material.id);
  }
  zoomInStart() {
    if (this.material == null) return;
    if (this.material.unit == null) return;
    this.material?.wsService.cameraZoomInStart(this.material.unit.id, this.material.id);
  }

  setState(valueState: CapabilityState): void {

  }
}
